### PR TITLE
doc: Update missing versioning documentation

### PR DESCRIPTION
**Pull request by vadmeste**
Remove stale 'lock' command as well.

### CODE REVIEW

**Review by kannappanr — APPROVED** [review]

File: docs/minio-client-complete-guide.md [hunk 1]
```diff
@@ -593,57 +623,29 @@ mc head -n 1 --encrypt-key "play/mybucket=32byteslongsecretkeymustbegiven1" play
 Hello!!
 ```
 
-<a name="lock"></a>
-### Command `lock`
-`lock` sets and gets object lock configuration
-
+*Example: Display the first lines of the content of an object, 1 year earlier*
```
Comment (eco-minio) [inline docs/minio-client-complete-guide.md:626]:
```suggestion
*Example: Display the first line of the content of an object, 1 year earlier*
```

File: docs/minio-client-complete-guide.md [hunk 2]
```diff
@@ -656,37 +658,56 @@ Removing `myminio/mybucket/prefix/comp.csv`.
...
 ```
 
+*Example: Disable legal hold on a specific object version*
+```
+mc legalhold clear myminio/mybucket/prefix/obj.csv --version-id "HiMFUTOowG6ylfNi4LKxD3ieHbgfgrvC"
+```
+
+*Example: Show object legal hold recursively for all objects at a prefix*
+```
+c legalhold info myminio/mybucket/prefix --recursive
```
Comment (eco-minio) [inline docs/minio-client-complete-guide.md:707]:
```suggestion
mc legalhold info myminio/mybucket/prefix --recursive
```

File: docs/minio-client-complete-guide.md [hunk 3]
```diff
@@ -793,6 +816,13 @@ mc cp -a myobject.txt play/mybucket
 myobject.txt:    14 B / 14 B  ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  100.00 % 41 B/s 0
 ```
 
+*Example: Roll back to object version of 10 days earlier while copying.*
```
Comment (eco-minio) [inline docs/minio-client-complete-guide.md:819]:
```suggestion
*Example: Roll back to object version to 10 days earlier while copying.*
```

**Review by eco-minio — CHANGES_REQUESTED** [review]
Few minor changes

File: docs/minio-client-complete-guide.md [hunk 4]
```diff
@@ -308,9 +310,9 @@ mc version RELEASE.2020-04-25T00-43-23Z
...
-| [**update** - manage software updates](#update)                                         | [**watch** - watch for events](#watch)                              | [**stat** - stat contents of objects and folders](#stat)   |                                                    |
-| [**head** - display first 'n' lines of an object](#head)                                | [**lock** - manage default bucket object lock configuration](#lock) | [**retention** - set retention for object(s)](#retention)  |                                                    |
-| [**mv** - move objects](#mv)                                                            | [**sql** - run sql queries on objects](#sql)                        | [**legalhold** - set legal hold for object(s)](#legalhold) |                                                    |
+| [**update** - manage software updates](#update)                                         | [**watch** - watch for events](#watch)                              | [**stat** - stat contents of objects and folders](#stat)   | [**du** - summarize disk usage recursively](#du)   |
```
[the remark this answers is not in the document]
Comment (vadmeste) [reply docs/minio-client-complete-guide.md:313]: > #du does not redirect to appropriate anchor in the page (works for tag and admin below)

yes, du documentation was just missing, fixed it now

**Comment by kannappanr** [comment]
@vadmeste Can you please resolve the conflicts. Will try to push this in quickly

**Comment by vadmeste** [comment]
> @vadmeste Can you please resolve the conflicts. Will try to push this in quickly

@kannappanr done!

**Review by eco-minio — APPROVED** [review]
LGTM

**Pull request merged**